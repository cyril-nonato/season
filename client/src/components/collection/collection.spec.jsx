import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Collection from './collection.component';
import '@testing-library/jest-dom/extend-expect';

jest.mock('../collection-item/collection-item.container', () => {
  const CollectionItemContainer = () => <div></div>;
  return CollectionItemContainer;
});

afterEach(cleanup);

describe('Collection Test', () => {
  let wrapper;
  const collection = {
    brand: 'some brand',
    items: [
      {
        id: 1,
        model: 'some model'
      }
    ]
  }
  beforeEach(() => {
    wrapper = render(<Collection collection={collection} />)
  });

  it('Should render the props', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('brand')).toHaveTextContent(collection.brand)
  });

  it('Should match the snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  })

});