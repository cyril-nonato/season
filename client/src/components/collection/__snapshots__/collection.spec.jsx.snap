// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Test Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa jXpelC"
  >
    <h3
      class="sc-bwzfXH bCkXwr"
      data-testid="brand"
    >
      some brand
    </h3>
    <ul
      class="sc-htpNat iCXrmo"
    >
      <div />
    </ul>
  </div>
</DocumentFragment>
`;
